WordPress/WordPressTest/Dashboard/FreeToPaidPlansDashboardCardHelperTests.swift

```diff
@@ -0,0 +1,69 @@
+import XCTest
+@testable import WordPress
+
+final class FreeToPaidPlansDashboardCardHelperTests: CoreDataTestCase {
+    func testShouldShowCardWithFreePlanAndNoMappedDomain() {
+        let blog = BlogBuilder(mainContext)
+            .with(supportsDomains: true)
+            .with(domainCount: 1, of: .wpCom)
+            .with(hasMappedDomain: false)
+            .with(hasPaidPlan: false)
+            .build()
+
+        let result = FreeToPaidPlansDashboardCardHelper.shouldShowCard(for: blog, isJetpack: true, featureFlagEnabled: true)
+
+        XCTAssertTrue(result, "Card should show for blogs with a free plan and a mapped domain")
+    }
+
+    func testShouldNotShowCardWithoutFreePlan() {
+        let blog = BlogBuilder(mainContext)
+            .with(supportsDomains: true)
+            .with(domainCount: 1, of: .wpCom)
+            .with(hasMappedDomain: false)
+            .with(hasPaidPlan: true)
+            .build()
+
+        let result = FreeToPaidPlansDashboardCardHelper.shouldShowCard(for: blog, isJetpack: true, featureFlagEnabled: true)
+
+        XCTAssertFalse(result, "Card should not show for blogs without a free plan")
+    }
+
+    func testShouldNotShowCardWithMappedDomain() {
+        let blog = BlogBuilder(mainContext)
+            .with(supportsDomains: true)
+            .with(domainCount: 1, of: .wpCom)
```

Comment:
```suggestion
            .with(domainCount: 1, of: .registered)
```

I expected either just a `.registered` domain or both a `.wpCom` and a `.registered` domain. I'm not sure if it makes a difference to the logic though. Wdyt?

Reply:
Yes, you're right.

That's a limitation of `BlogBuilder` since some properties are computed and some stored. 

`hasMappedDomain` uses stored options to determine whether a mapped domain exists. In reality, it would have a connection with the number of registered domains. In the testing environment, however, there's no connection.

Reply:
Gotcha, marking as resolved!